### PR TITLE
[ES|QL] Use describe method to generate OrdinalGroupingOperator profile message

### LINKED ISSUE #111674: ESQL: Ordinals grouping operator's status has default toStrings in it
### Description

The status and profile for when ESQL groups on ordinals looks like:
```
{"operator":"OrdinalsGroupingOperator[aggregators=[org.elasticsearch.compute.aggregation.AggregatorFunctionSupplier$2@798e0d19, org.elasticsearch.compute.aggregation.AggregatorFunctionSupplier$2@f58d3d6]]"}
```

It should look more like:
```
Aggregator[aggregatorFunction=CountAggregatorFunction[channels=[1]
```

We're letting the toString for a closure slip in to something we're using for debug output.

I used the nyc_taxis data set from rally to get this and ran:
```
curl -s -uelastic:password -HContent-Type:application/json localhost:9200/_query?pretty -d'{
  "query": "FROM nyc_taxis | INLINESTATS avg=AVG(total_amount) BY payment_type | KEEP total_amount, avg, payment_type | WHERE total_amount > avg | LIMIT 2",
	"pragma": {
		"data_partitioning": "shard"
	},
	"profile": true
}' | jq -c .profile.drivers[]
```

### PR DESCRIPTION
**Pull request by drewdaemon**
Close https://github.com/elastic/elasticsearch/issues/111674

**Before**

`OrdinalsGroupingOperator[aggregators=[org.elasticsearch.compute.aggregation.AggregatorFunctionSupplier$2@5ccd8c4, org.elasticsearch.compute.aggregation.AggregatorFunctionSupplier$2@460e88c2]]`

**After**

`OrdinalsGroupingOperator[aggregators=["sum of longs", "count"]]`

### CODE REVIEW

File: x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java [hunk 1]
```diff
@@ -323,6 +323,35 @@ public void testProfile() throws IOException {
...
+            builder.pragmas(Settings.builder().put("data_partitioning", "shard").build());
+        }
+        Map<String, Object> result = runEsql(builder);
+
+        List<List<String>> signatures = new ArrayList<>();
+        @SuppressWarnings("unchecked")
+        List<Map<String, Object>> profiles = (List<Map<String, Object>>) ((Map<String, Object>) result.get("profile")).get("drivers");
+        for (Map<String, Object> p : profiles) {
+            fixTypesOnProfile(p);
+            assertThat(p, commonProfile());
```
Comment (drewdaemon) [inline x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java:342]: Not sure if we want these checks in this test as well...

**Comment by drewdaemon** [comment]
@elasticsearchmachine merge upstream

**Comment by drewdaemon** [comment]
@nik9000 I don't really understand these CI failures. Could you take a look?

File: x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java:342]: It should be ok.

**Comment by nik9000** [comment]
> @nik9000 I don't really understand these CI failures. Could you take a look?

oh no. I was just looking finally and you pushed and they are gone. I'll try and look once it's done again.

**Review by nik9000 — APPROVED** [review]

**Comment by drewdaemon** [comment]
Sorry @nik9000 ! 😅

**Comment by nik9000** [comment]
OK! Failure's back:
```
:x-pack:plugin:transform:qa:multi-cluster-tests-with-security:mixed-cluster
...
process was found dead while waiting for ports files
```

*Generally* something like this is because your branch is out of date with `main`. The transport version in `8.x` is backwards compatible `main`, but not with a `main` forked a week ago. Or a month ago. Or whatever. So if you merge `main` into this branch and push again it *should* vanish. Try that first before thinking too hard about what might be up with that error.

**Comment by drewdaemon** [comment]
@nik9000 I'm up-to-date with `main`, but it looks like I'm still getting that error 🤔

**Pull request merged**